llpc/context/llpcPipelineContext.cpp

```diff
@@ -616,10 +616,10 @@ ShaderOptions PipelineContext::computeShaderOptions(const PipelineShaderInfo &sh
   if (ScalarizeWaterfallDescriptorLoads.getNumOccurrences() > 0) {
     shaderOptions.scalarizeWaterfallLoads = ScalarizeWaterfallDescriptorLoads;
   } else {
-    shaderOptions.scalarizeWaterfallLoads = shaderInfo.options.scalarizeWaterfallLoads;
-    // Enable waterfall load scalarization when vgpr limit is set.
-    if (shaderOptions.vgprLimit != 0 && shaderOptions.vgprLimit != UINT_MAX)
-      shaderOptions.scalarizeWaterfallLoads = true;
+    shaderOptions.scalarizeWaterfallLoads = true;
```

Comment:
I'd prefer something like (if you keep the optional):

```
if (shaderInfo.options.scalarizeWaterfallLoads.has_value())
    shaderOptions.scalarizeWaterfallLoads = shaderInfo.options.scalarizeWaterfallLoads.value();
else
    shaderOptions.scalarizeWaterfallLoads = true;
```

Reply:
Maybe `shaderOptions.scalarizeWaterfallLoads = shaderInfo.options.scalarizeWaterfallLoads.value_or(true)`?

Reply:
Done as Thomas suggested :)